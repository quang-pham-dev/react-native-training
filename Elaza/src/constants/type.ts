export const SIZE = {
  TINY: 'tiny',
  SMALL: 'small',
  MEDIUM: 'medium',
  LARGE: 'large',
  BASE: 'base',
}

export const HEADING_TYPE = {
  H1: 'h1',
  H2: 'h2',
  H3: 'h3',
  H4: 'h4',
}

export const PARAGRAPH_TYPE = {
  TINY: 'tiny',
  SMALL: 'small',
  BASE: 'base',
  MEDIUM: 'medium',
  LARGER: 'larger',
}

export const BUTTON_TYPE = {
  CONFIRM: 'confirm',
  CANCEL: 'cancel',
  LINK: 'link',
  SUBMIT: 'submit',
}
