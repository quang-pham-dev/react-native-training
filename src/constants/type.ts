export const SIZE = {
  TINY: 'tiny',
  SMALL: 'small',
  MEDIUM: 'medium',
  LARGE: 'large',
  BASE: 'base',
}

export const HEADING_TYPE = {
  H1: 'h1',
  H2: 'h2',
  H3: 'h3',
  H4: 'h4',
}

export const PARAGRAPH_TYPE = {
  TINY: 'tiny',
  SMALL: 'small',
  BASE: 'base',
  MEDIUM: 'medium',
  LARGER: 'larger',
  BRAND: 'brand',
  PRODUCT_NAME_AND_TYPE: 'productNameAndType',
  PRODUCT_PRICE: 'productPrice',
  ALREADY_ACCOUNT: 'alreadyAccount',
  ALREADY_SIGN_IN: 'alreadySignIn',
  CONFIRM_AGREE: 'confirmAgree',
  TERM_AND_CONDITION: 'termAndCondition',
  PROFILE_USERNAME: 'profileUsername',
  PROFILE_VERIFIED: 'profileVerified',
  PROFILE_ITEM: 'profileItem',
  BRAND_DETAIL_TITLE: 'brandDetailTitle',
  BRAND_DETAIL_ITEM_COUNT: 'brandDetailItemCount',
}

export const BUTTON_TYPE = {
  BOTTOM: 'bottom',
  SOCIAL: 'social',
}
