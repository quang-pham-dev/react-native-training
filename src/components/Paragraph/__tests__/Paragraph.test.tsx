import React from 'react'
import {create} from 'react-test-renderer'

import ParagraphStyled from '../P.styles'

// Constants
import {PARAGRAPH_TYPE} from '@constants/type'

describe('Paragraph', () => {
  test('should render Paragraph normal with tiny text correctly', () => {
    const component = create(
      <ParagraphStyled.Base type={PARAGRAPH_TYPE.TINY}>
        Paragraph
      </ParagraphStyled.Base>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should render Paragraph normal with base text correctly', () => {
    const component = create(
      <ParagraphStyled.Base type={PARAGRAPH_TYPE.BASE}>
        Paragraph
      </ParagraphStyled.Base>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should render Paragraph normal with small text correctly', () => {
    const component = create(
      <ParagraphStyled.Base type={PARAGRAPH_TYPE.SMALL}>
        Paragraph
      </ParagraphStyled.Base>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should render Paragraph normal with medium text correctly', () => {
    const component = create(
      <ParagraphStyled.Base type={PARAGRAPH_TYPE.MEDIUM}>
        Paragraph
      </ParagraphStyled.Base>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should render Paragraph normal with large text correctly', () => {
    const component = create(
      <ParagraphStyled.Base type={PARAGRAPH_TYPE.LARGER}>
        Paragraph
      </ParagraphStyled.Base>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should render Paragraph center correctly', () => {
    const component = create(
      <ParagraphStyled.Center type={PARAGRAPH_TYPE.MEDIUM}>
        Paragraph
      </ParagraphStyled.Center>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should render Paragraph Right correctly', () => {
    const component = create(
      <ParagraphStyled.Right type={PARAGRAPH_TYPE.MEDIUM}>
        Paragraph
      </ParagraphStyled.Right>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should render Paragraph font family base correctly', () => {
    const component = create(
      <ParagraphStyled.TFamilyRegular type={PARAGRAPH_TYPE.MEDIUM}>
        Paragraph
      </ParagraphStyled.TFamilyRegular>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should render Paragraph font family bold correctly', () => {
    const component = create(
      <ParagraphStyled.TFamilyBold type={PARAGRAPH_TYPE.MEDIUM}>
        Paragraph
      </ParagraphStyled.TFamilyBold>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should render Paragraph font family medium correctly', () => {
    const component = create(
      <ParagraphStyled.TFamilySemiBold type={PARAGRAPH_TYPE.MEDIUM}>
        Paragraph
      </ParagraphStyled.TFamilySemiBold>,
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
