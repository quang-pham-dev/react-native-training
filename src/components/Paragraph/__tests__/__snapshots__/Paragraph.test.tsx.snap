// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph should render Paragraph Right correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#1D1E20",
        "fontFamily": "Inter-Regular",
        "fontSize": 17,
        "lineHeight": 16,
        "textAlign": "right",
      },
    ]
  }
  type="medium"
>
  Paragraph
</Text>
`;

exports[`Paragraph should render Paragraph center correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#1D1E20",
        "fontFamily": "Inter-Regular",
        "fontSize": 17,
        "lineHeight": 16,
        "textAlign": "center",
      },
    ]
  }
  type="medium"
>
  Paragraph
</Text>
`;

exports[`Paragraph should render Paragraph font family base correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#1D1E20",
        "fontFamily": "Inter-Regular",
        "fontSize": 17,
        "lineHeight": 16,
        "textAlign": "left",
      },
    ]
  }
  type="medium"
>
  Paragraph
</Text>
`;

exports[`Paragraph should render Paragraph font family bold correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#1D1E20",
        "fontFamily": "Inter-SemiBold",
        "fontSize": 17,
        "lineHeight": 16,
        "textAlign": "left",
      },
    ]
  }
  type="medium"
>
  Paragraph
</Text>
`;

exports[`Paragraph should render Paragraph font family medium correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#1D1E20",
        "fontFamily": "Inter-Medium",
        "fontSize": 17,
        "lineHeight": 16,
        "textAlign": "left",
      },
    ]
  }
  type="medium"
>
  Paragraph
</Text>
`;

exports[`Paragraph should render Paragraph normal with base text correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "textAlign": "left",
      },
    ]
  }
  type="base"
>
  Paragraph
</Text>
`;

exports[`Paragraph should render Paragraph normal with large text correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#1D1E20",
        "fontFamily": "Inter-SemiBold",
        "fontSize": 22,
        "lineHeight": 22,
        "textAlign": "left",
      },
    ]
  }
  type="larger"
>
  Paragraph
</Text>
`;

exports[`Paragraph should render Paragraph normal with medium text correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#1D1E20",
        "fontFamily": "Inter-Regular",
        "fontSize": 17,
        "lineHeight": 16,
        "textAlign": "left",
      },
    ]
  }
  type="medium"
>
  Paragraph
</Text>
`;

exports[`Paragraph should render Paragraph normal with small text correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#1D1E20",
        "fontFamily": "Inter-Regular",
        "fontSize": 15,
        "lineHeight": 11,
        "textAlign": "left",
      },
    ]
  }
  type="small"
>
  Paragraph
</Text>
`;

exports[`Paragraph should render Paragraph normal with tiny text correctly 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#1D1E20",
        "fontFamily": "Inter-Regular",
        "fontSize": 11,
        "lineHeight": 11,
        "textAlign": "left",
      },
    ]
  }
  type="tiny"
>
  Paragraph
</Text>
`;
